stablehlo/dialect/StablehloOps.td

```diff
@@ -2084,6 +2084,44 @@ def StableHLO_CollectivePermuteOp: StableHLO_Op<"collective_permute",
       "::mlir::DenseIntElementsAttr":$source_target_pairs)>];
 }
 
+def StableHLO_CompositeOp : StableHLO_Op<"composite", []> {
```

Comment:
How about implementing `SymbolUserOpInterface`:

```
def CallOp : Func_Op<"call",
    [CallOpInterface, MemRefsNormalizable,
     DeclareOpInterfaceMethods<SymbolUserOpInterface>]>
```

Makes the op have a method called:

```
LogicalResult CompositeOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
   hlo::verifyCompositeOp(<other_fields>, symbolTable);
}
```

We can do all verification in this op, including sym validation.

Reply:
Good callout! I did think to use `SymbolUserOpInterface`, but I'm worried it might have unintended side effects, e.g. it seems it will prevent running the `RemoveDeadValues` pass, unless we also implement `CallOpInterface`: https://github.com/llvm/llvm-project/blob/f51ade25b9205efee09a4915031848cebe772805/mlir/lib/Transforms/RemoveDeadValues.cpp#L581

`composite` isn't actually a call, so I don't think that makes sense, but I don't know if we care about the dead values thing.

I also don't think we gain a whole lot by implementing this interface. If you feel strongly about this I'm happy to do it, otherwise maybe this can be left as something to investigate in the future?

Reply:
Do we use this pass? There was one dead-code-removal pass that didn't work for us for various reasons as-is. Something that wasn't CSE or Canonicalization-related cleanups.

That said, no strong opinion. It buys us a guarantee that the properly scoped symbol table will be passed to us even if upstream MLIR changes anything. That said if it ever became different from the sym table of the module there would probably be a lot of cleanup to do.

Reply:
We don't, but I wonder if downstream users might.

In any case, I think you make a good point. And it's probably best to signal that we use a symbol. Maybe we should do this for `custom_call` too.

I'll make the change tomorrow.